Description: Update Sage's planarity module to use API v3
 Sage uses 2.2.0 but the API difference from planarity 3 is not too bad.
 So let's try this instead of backporting planarity 2.2.0 to Debian.
Author: Ximin Luo <infinity0@debian.org>
Bug: TODO
Forwarded: TODO
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/sage/src/sage/graphs/planarity.pyx
+++ b/sage/src/sage/graphs/planarity.pyx
@@ -4,12 +4,12 @@
 
 cdef extern from "planarity/graph.h":
     ctypedef struct graphNode:
-        int v
+        int index
         int link[2]
     ctypedef graphNode * graphNodeP
 
     ctypedef struct BM_graph:
-        graphNodeP G
+        graphNodeP V
         int N
     ctypedef BM_graph * graphP
 
@@ -136,10 +136,10 @@
         from sage.graphs.graph import Graph
         for i from 0 <= i < theGraph.N:
             linked_list = []
-            j = theGraph.G[i].link[1]
+            j = theGraph.V[i].link[1]
             while j >= theGraph.N:
-                linked_list.append(to[theGraph.G[j].v])
-                j = theGraph.G[j].link[1]
+                linked_list.append(to[theGraph.V[j].index])
+                j = theGraph.V[j].link[1]
             if len(linked_list) > 0:
                 g_dict[to[i]] = linked_list
         G = Graph(g_dict)
@@ -155,10 +155,10 @@
                 #for i in range(theGraph.N):
                 for i from 0 <= i < theGraph.N:
                     linked_list = []
-                    j = theGraph.G[i].link[1]
+                    j = theGraph.V[i].link[1]
                     while j >= theGraph.N:
-                        linked_list.append(to[theGraph.G[j].v])
-                        j = theGraph.G[j].link[1]
+                        linked_list.append(to[theGraph.V[j].index])
+                        j = theGraph.V[j].link[1]
                     emb_dict[to[i]] = linked_list
                 g._embedding = emb_dict
             if set_pos:
@@ -176,10 +176,10 @@
                 #for i in range(theGraph.N):
                 for i from 0 <= i < theGraph.N:
                     linked_list = []
-                    j = theGraph.G[i].link[0]
+                    j = theGraph.V[i].link[0]
                     while j >= theGraph.N:
-                        linked_list.append(to[theGraph.G[j].v])
-                        j = theGraph.G[j].link[0]
+                        linked_list.append(to[theGraph.V[j].index])
+                        j = theGraph.V[j].link[0]
                     emb_dict[to[i]] = linked_list
                 g._embedding = emb_dict
         gp_Free(&theGraph)
